jupyter_server_fileid/manager.py

```diff
@@ -37,35 +37,174 @@ def wrapped(self, *args, **kwargs):
+    unique ID. The path is only updated upon calling `move()`, `copy()`, or
+    `delete()`, e.g. upon receipt of contents manager events emitted by Jupyter
+    Server 2.
+    """
+
+    def __init__(self, *args, **kwargs):
+        # pass args and kwargs to parent Configurable
+        super().__init__(*args, **kwargs)
+        # initialize instance attrs
+        self._update_cursor = False
+        # initialize connection with db
+        self.log.info(f"ArbitraryFileIdManager : Configured root dir: {self.root_dir}")
+        self.log.info(f"ArbitraryFileIdManager : Configured database path: {self.db_path}")
+        self.con = sqlite3.connect(self.db_path)
+        self.log.info("ArbitraryFileIdManager : Successfully connected to database file.")
+        self.log.info("ArbitraryFileIdManager : Creating File ID tables and indices.")
+        # do not allow reads to block writes. required when using multiple processes
+        self.con.execute("PRAGMA journal_mode = WAL")
+        self.con.execute(
+            "CREATE TABLE IF NOT EXISTS Files("
+            "id INTEGER PRIMARY KEY AUTOINCREMENT, "
+            "path TEXT NOT NULL UNIQUE"
+            ")"
+        )
+        self.con.execute("CREATE INDEX IF NOT EXISTS ix_Files_path ON Files (path)")
+        self.con.commit()
+
+    def index(self, path: str) -> int:
+        row = self.con.execute("SELECT id FROM Files WHERE path = ?", (path,)).fetchone()
+        existing_id = row and row[0]
+
+        if existing_id:
+            return existing_id
+
+        # create new record
+        cursor = self.con.execute("INSERT INTO Files (path) VALUES (?)", (path,))
+        self.con.commit()
+        return cursor.lastrowid  # type:ignore
+
+    def get_id(self, path: str) -> Optional[int]:
+        row = self.con.execute("SELECT id FROM Files WHERE path = ?", (path,)).fetchone()
+        return row and row[0]
+
+    def get_path(self, id: Union[int, str]) -> Optional[int]:
+        row = self.con.execute("SELECT path FROM Files WHERE id = ?", (id,)).fetchone()
+        return row and row[0]
+
+    def move(self, old_path: str, new_path: str) -> None:
+        row = self.con.execute("SELECT id FROM Files WHERE path = ?", (old_path,)).fetchone()
+        id = row and row[0]
+
+        if id:
+            self.con.execute("UPDATE Files SET path = ? WHERE path = ?", (new_path, old_path))
+        else:
+            cursor = self.con.execute("INSERT INTO Files (path) VALUES (?)", (new_path,))
+            id = cursor.lastrowid
+
+        self.con.commit()
+        return id
+
+    def copy(self, from_path: str, to_path: str) -> Optional[int]:
+        cursor = self.con.execute("INSERT INTO Files (path) VALUES (?)", (to_path,))
+        self.con.commit()
+        return cursor.lastrowid
+
+    def delete(self, path: str) -> None:
+        self.con.execute("DELETE FROM Files WHERE path = ?", (path,))
+        self.con.commit()
+
+    def save(self, path: str) -> None:
+        return
```

Comment:
This should be implemented to call `_index()` if its not already indexed.  Otherwise, no entries will ever be inserted into the FILES table.

Also, `_get()` should be implemented such that it should be called on `Get` Contents events.  Similar reason to why `Save` events must be handled.